src/hotspot/share/opto/ifnode.cpp

```diff
@@ -1927,6 +1951,11 @@ Node* RangeCheckNode::Ideal(PhaseGVN *phase, bool can_reshape) {
       // the interpreter.  If we see range-check deopt's, do not widen!
       if (!phase->C->allow_range_check_smearing())  return nullptr;
 
+      if (can_reshape && !phase->C->post_loop_opts_phase()) {
+        phase->C->record_for_post_loop_opts_igvn(this);
+        return nullptr;
+      }
+
```

Comment:
Why is it ok to delay this to post-loop-opts? Does it not prevent moving some CFG from being eliminated out of loops? Would be nice to have a little justification comment.

Reply:
Ah. Does this mean that if there are multiple RangeCheck in a loop, where some could be smeared, these are not smeared, and then we have more RangeChecks to eliminate out of the loop? Maybe in the end this all comes down to the same anyway. Just wondering.

Reply:
@rwestrel would you mind explaining exactly that in a comment? Something like:
```
We are about to perform range check smearing (i.e. remove this RangeCheck if it is dominated
by two RangeChecks which have a range that covers the this RangeCheck). This can cause
nodes to be pinned. We want to avoid that and first allow RangeCheckElimination a chance to
remove the RangeChecks from loops. Hence, we delay range check smearing until after loop opts.
```